RFP-01.md

```diff
@@ -0,0 +1,212 @@
+# RFP-01: Settlement contract wrapper
+
+## Preamble
+
+Requests for proposals are not to be taken as prescriptive or exhaustive. The community is encouraged to submit proposals that build upon the ideas presented in this document. The scope of the project may change based on the proposal received. The primary intent of this document is to provide a starting point from which to achieve the goals outlined, and what is ultimately implemented may differ from the initial proposal.
+
+## Background
+
+The CoW DAO has recently passed [CIP-44: Reduced bonding requirements](https://snapshot.org/#/cow.eth/proposal/0x1b6f1171633ec3d20c4370db37074aa1bd830486d4d0d6c26165915cc42d9412), which allows solvers to participate in the batch auction with reduced bonding requirements. The reduced bonding requirements are as follows:
+
+- $50k in yield-bearing stable coins or ETH
+- 500k COW
+
+The target deposit is:
+
+- $100k in yield-bearing stable coins or ETH
+- 1M COW
+
+## Goal
+
+The goal of this project is to implement a `SubPoolFactory` that allows solvers to participate in the batch auction with reduced bonding requirements. The `SubPoolFactory` will deploy a `SubPool` for each solver, which will be funded to at least the minimum required amount. The `SubPoolFactory` will also provide a wrapper around the `GPv2Settlement` contract that allows solvers to execute their solutions by providing a signature attesting to the solution and their parameters.
+
+### Key Features
+
+- Ultra gas efficient signed settlement function
+- No co-mingling of funds between solvers
+- Easily observable on-chain addresses for MEV Blocker rebates
+- Easier access to the batch auction with reduced bonding requirements
+
+### Deliverables
+
+- Contract system for `SubPoolFactory`, `SubPool`, and `SignedSettlement`
+- Comprehensive documentation
+- Open-source code repository
+
+### Interface
+
+The interfaces below are a _starting point_ for the project and are not exhaustive / binding - they have not be audited or exhaustively tested / reasoned. The community is encouraged to submit proposals that build upon the ideas presented in this document.
+
+```solidity
+interface Auth {
+    /**
+     * Allows `usr` to perform root-level functions
+     */
+    function addOwner(address usr) external auth;
+    /**
+     * Denies `usr` from being able to perform root-level functions
+     */
+    function removeOwner(address usr) external auth;
+    /**
+     * Determine if `usr` is able to perform root-level functions
+     */
+    function isOwner(address usr) external view returns (bool);
+}
+
+interface SubPool is Auth {
+    /**
+     * Determine the amount required to `heal` the pool
+     * @return amt of immutable collateralToken
+     * @return cowAmt of COW
+     */
+    function dues() external view returns (uint256 amt, uint256 cowAmt);
```

Comment:
without minimum configuration for collateral amounts, this can only return difference in cow amounts from minimum amount.

Reply:
I think this should return the damage that has been invoiced (which doesn't necessarily need to know the minimal amounts)